deployer/deployer-services/src/main/java/io/hyscale/deployer/services/deployer/impl/KubernetesDeployer.java

```diff
@@ -369,13 +388,74 @@ public static void waitForPodActivity(ApiClient apiClient, V1PodHandler podHandl
             throw new HyscaleException(DeployerErrorCodes.FAILED_TO_INITIALIZE_POD);
         }
     }
-
-    private void deltaWait(long milliSeconds) {
-        try {
-            Thread.sleep(milliSeconds);
-        } catch (InterruptedException e) {
+    
+	public void watch(ApiClient apiClient, String appName, String serviceName, String namespace)
+			throws IOException, ApiException, HyscaleException {
+		OkHttpClient httpClient = apiClient.getHttpClient().newBuilder().readTimeout(0, TimeUnit.SECONDS).build();
+		apiClient.setHttpClient(httpClient);
+		Configuration.setDefaultApiClient(apiClient);
+		CoreV1Api api = new CoreV1Api();
+		Set<String> pods = new HashSet<String>();
+		String selector = ResourceSelectorUtil.getServiceSelector(appName, serviceName);
+		Integer replicas = null;
+		V1DeploymentHandler v1DeploymentHandler = new V1DeploymentHandler();
+		V1Deployment v1Deployment=null;
+		try {
+			v1Deployment = v1DeploymentHandler.get(apiClient, serviceName, namespace);
+			replicas=v1Deployment.getSpec().getReplicas();
+		}catch(Exception e) {
+			replicas=1;
+		}	
+		Watch<V1Pod> watch = null;
+		String namespaceSelector=ResourceSelectorUtil.getNamespaceSelector(namespace);
+		watch = Watch.createWatch(apiClient,
```

Comment:
Watch should have an extra label based on Deployment or StatefulSet
controller-revision-hash or pod-template-hash...try to fetch the label and filter pods at Watch itself.

Reply:
Have a look at PodParentHandler...there is method called getPodSelector ...this is based on the parent of the pod.
It's just that you need to figure out the parent of the pod is deployment of statefulset.
and call this method and merge the labels 
and then watch